### ISSUE TITLE
shared_image_gallery_replica_count value ignored with v2.0.0

### ISSUE THREAD

**Issue by raptorrico** [hashicorp/packer-plugin-azure#341]
#### Overview of the Issue

When running v2.0.0, the int value of 'shared_image_gallery_replica_count' is ignored. Resulting image version always has a replica count of 1.

Reverting to version 1.4.5 resolves the issue.

#### Reproduction Steps

```
packer {
  required_plugins {
    azure = {
      version = ">= 2.0.0"
      source  = "github.com/hashicorp/azure"
    }
  }
}
```

### Plugin and Packer version

2.0.0 and 1.9.4

### Simplified Packer Buildfile

```
variable "gallery_destination_rg"           {}
variable "gallery_image_name"               {}
variable "gallery_image_version"            {}
variable "gallery_name"                     {}
variable "gallery_subscription"             {}
variable "managed_image_name"               {}
variable "managed_image_resource_group"     {}
variable "tags"                             {}
variable "temp_compute_name"                {}
variable "tenant_id"                        {}
variable "user_assigned_managed_identities" {}
variable "virtual_network_name"             {}
variable "virtual_network_subnet_name"      {}

source "azure-arm" "win11" {
    azure_tags                        = var.tags
    tenant_id                         = var.tenant_id
    use_azure_cli_auth                = true
    user_assigned_managed_identities  = var.user_assigned_managed_identities
    image_offer                       = "office-365"
    image_publisher                   = "MicrosoftWindowsDesktop"
    image_sku                         = "win11-22h2-avd-m365"
    location                          = "uksouth"
    os_type                           = "Windows"
    vm_size                           = "Standard_D2_v4"

    shared_image_gallery_destination {
        gallery_name                  = var.gallery_name
        image_name                    = var.gallery_image_name
        image_version                 = var.gallery_image_version
        replication_regions           = ["uksouth"]
        resource_group                = var.gallery_destination_rg
        storage_account_type          = "Standard_LRS"
        subscription                  = var.gallery_subscription
    }
    virtual_network_name                              = var.virtual_network_name
    virtual_network_subnet_name                       = var.virtual_network_subnet_name
    shared_image_gallery_replica_count                = 3
    shared_gallery_image_version_end_of_life_date     = timeadd(timestamp(), "1440h")
    shared_gallery_image_version_exclude_from_latest  = false
}

packer {
  required_plugins {
    windows-update = {
      version = ">= 0.14.3"
      source  = "github.com/rgl/windows-update"
    }
    azure = {
      version = ">= 1.4.1"
      source  = "github.com/hashicorp/azure"
    }
  }
}

build {
    sources    = ["source.azure-arm.win11"]
    provisioner "windows-shell" {
        inline = ["mkdir c:\\test"]
    }
    provisioner "file" {
        source  = "../../scripts"
        destination = "c:/test"
    }
}
```

### Operating system and Environment details

Windows 11 multi-session

### Log Fragments and crash.log files

Unable to supply.

**Comment by raptorrico** [comment]
Hello, when can we expect the replication count bug to be fixed? Thanks

**Comment by JenGoldstrich** [comment]
Hey @raptorrico thanks for reporting this, this issue has been resolved in v2.0.1 of the plugin which was released today.
